kotlinx-coroutines-core/common/src/flow/operators/Lint.kt

```diff
@@ -79,4 +80,61 @@ public fun FlowCollector<*>.cancel(cause: CancellationException? = null): Unit =
     replaceWith = ReplaceWith("currentCoroutineContext()")
 )
 public val FlowCollector<*>.coroutineContext: CoroutineContext
-    get() = noImpl()
\ No newline at end of file
+    get() = noImpl()
+
+@Deprecated(
+    message = "SharedFlow never completes, so this operator has no effect.",
+    level = DeprecationLevel.WARNING,
+    replaceWith = ReplaceWith("this")
+)
+@InlineOnly
+public inline fun <T> SharedFlow<T>.catch(noinline action: suspend FlowCollector<T>.(cause: Throwable) -> Unit): Flow<T> =
+    (this as Flow<T>).catch(action)
+
+@Deprecated(
+    message = "SharedFlow never completes, so this operator has no effect.",
```

Comment:
maybe text ressource move to const? Any optimize  f for equal annotation dubles?

Reply:
What do you mean?

Reply:
No way to reduce copy-paste / duplicate code / messages?

Reply:
There is no reason to. The code is easier to read this way, with messages right here.